From cefa8e80a7210fb13cf915765c1ba72e25ac614f Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Mon, 3 Nov 2014 07:52:14 -0500
Subject: [infiniband] iser: Fix sector_t format warning

Message-id: <6ae6420250883cb6f5cd22c957fae2b5da8b4a10.1415000301.git.avadai@redhat.com>
Patchwork-id: 98685
O-Subject: [Patch RHEL7.1 16/37] IB/iser: Fix sector_t format warning
Bugzilla: 1107622
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Doug Ledford <dledford@redhat.com>

From: Randy Dunlap <rdunlap@infradead.org>

Fix pr_err (printk) format warning:

    drivers/infiniband/ulp/iser/iser_verbs.c:1181:4: warning: format '%lx' expects argument of type 'long unsigned int', but argument 3 has type 'sector_t' [-Wformat]

Signed-off-by: Randy Dunlap <rdunlap@infradead.org>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 39c978cd1704dc03f6913bd40f858def7fd31185)
Signed-off-by: Amir Vadai <amirv@mellanox.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/ulp/iser/iser_verbs.c b/drivers/infiniband/ulp/iser/iser_verbs.c
index abbb6ec..d2848e4 100644
--- a/drivers/infiniband/ulp/iser/iser_verbs.c
+++ b/drivers/infiniband/ulp/iser/iser_verbs.c
@@ -1178,9 +1178,10 @@ u8 iser_check_task_pi_status(struct iscsi_iser_task *iser_task,
    do_div(sector_off, sector_size + 8);
    *sector = scsi_get_lba(iser_task->sc) + sector_off;
 
-   pr_err("PI error found type %d at sector %lx "
+   pr_err("PI error found type %d at sector %llx "
           "expected %x vs actual %x\n",
-          mr_status.sig_err.err_type, *sector,
+          mr_status.sig_err.err_type,
+          (unsigned long long)*sector,
           mr_status.sig_err.expected,
           mr_status.sig_err.actual);
 
-- 
1.7.1